services/src/main/java/org/keycloak/models/policy/ResourcePolicyManager.java

```diff
@@ -317,4 +356,54 @@ private ComponentModel getPolicyComponent(String id) {
 
         return component;
     }
+
+    public ResourcePolicyRepresentation toRepresentation(ResourcePolicy policy) {
+        ResourcePolicyRepresentation rep = new ResourcePolicyRepresentation(policy.getId(), policy.getProviderId(), policy.getConfig());
+
+        for (ResourceAction action : getActions(policy.getId())) {
+            rep.addAction(toRepresentation(action));
+        }
+
+        return rep;
+    }
+
+    private ResourcePolicyActionRepresentation toRepresentation(ResourceAction action) {
+        List<ResourcePolicyActionRepresentation> actions = action.getActions().stream().map(this::toRepresentation).toList();
+        return new ResourcePolicyActionRepresentation(action.getId(), action.getProviderId(), action.getConfig(), actions);
+    }
+
+    public ResourcePolicy toModel(ResourcePolicyRepresentation rep) {
+        ResourcePolicyManager manager = new ResourcePolicyManager(session);
+        MultivaluedHashMap<String, String> config = ofNullable(rep.getConfig()).orElse(new MultivaluedHashMap<>());
+
+        for (ResourcePolicyConditionRepresentation condition : rep.getConditions()) {
+            String conditionProviderId = condition.getProviderId();
+            config.computeIfAbsent("conditions", key -> new ArrayList<>()).add(conditionProviderId);
+
+            for (Entry<String, List<String>> configEntry : condition.getConfig().entrySet()) {
+                config.put(conditionProviderId + "." + configEntry.getKey(), configEntry.getValue());
+            }
+        }
+
+        ResourcePolicy policy = manager.addPolicy(rep.getProviderId(), config);
+        List<ResourceAction> actions = new ArrayList<>();
+
+        for (ResourcePolicyActionRepresentation actionRep : rep.getActions()) {
```

Comment:
We should probably have some validation to prevent actions that do not support sub actions (not aggregated) from having them.

I can do that as a follow-up.